utils/filecache/interfaces.go

```diff
@@ -0,0 +1,42 @@
+package filecache
+
+import (
+	"context"
+	"io"
+
+	"github.com/ARM-software/golang-utils/utils/filesystem"
+)
+
+// IFileCache provides a filesystem cache for files and directories.
+// Cache entries are identified by a key and live for a configurable TTL (time-to-live).
+// An internal garbage collector periodically evicts expired entries; the period is also configurable.
+type IFileCache interface {
+	io.Closer
+
+	// Calls the entry provider to store an entry into the cache under the provided `key`, and starts the expiration
+	// timer based on the configured TTL.
+	// Returns an error if the path doesn't exist, the key already exists in the cache, the copy process fails or
+	// the cache is closed.
+	Store(ctx context.Context, key string) error
```

Comment:
would it be possible to also add a method in which we can specify a ttl for a particular entry:
e.g. StoreWithTTL(ctx context, key string, ttl time.Duration)